src/openforms/submissions/status.py

```diff
@@ -26,34 +29,53 @@ def get_async_result(self) -> Optional[AsyncResult]:
         return self._async_result
 
     @property
-    def state(self) -> str:
+    def status(self) -> str:
         result = self.get_async_result()
-        return result.state if result else ""
+        is_ready = result is not None and result.state in states.READY_STATES
+        return ProcessingStatuses.done if is_ready else ProcessingStatuses.in_progress
 
     @property
-    def processing_aborted(self) -> bool:
-        return self.state == states.FAILURE  # TODO: verify
+    def result(self) -> str:
+        result = self.get_async_result()
+        if result is None or result.state in states.UNREADY_STATES:
+            return ""
+
+        if result.state == states.SUCCESS:
+            return ProcessingResults.success
+        if result.state == states.REVOKED:
```

Comment:
no, that's `FAILURE`.

The table is nice, but doesn't actually seem to map to the real world, since `REVOKED` is not actually possible to be triggered at the moment. I wanted to discuss that at some point. The nuance between `retry` and `failed` is now not possible, I think.

Reply:
Did some more testing about this, with an incomplete appointment. At the moment the behaviour is not at all what we want.

1. The task in the chain does raise an (unrecoverable) exception, as intended
2. This however, does not cause the entire chain to be in the error state, it seems to just hang in the `PENDING` state.

I'm trying to figure out if I can get some insight into the sub-tasks from the chain task ID or not, but potentially we'll just have to store all the task IDs on the submissions in an array and check all of them to determine the "total" state.

Revoking tasks is also a complex matter - it's not something we can actually properly use. It only works for tasks that aren't executing yet (aborting is meant for that, but comes with it's own slew of unreliabilities). Revoking works by keeping a list of task IDs in memory (or file-system) with every worker and if a task gets scheduled to the worker, they consult if it's revoked or not and only then decide to execute it. This won't work for an already-running chain.

I propose to not block this PR any further on these details - I need to fix the state handling internally anyway, but want to do that in a separate PR so I can keep it focused. I'm confident that the external API will not change because of that, just the implementation details.